File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let counter = new Map;
        for(let num of nums) {
            if(counter.has(num)) counter.set(num, counter.get(num) + 1);
            else counter.set(num, 1);
        }
        let i = 0;
        let response = [];
        console.log(counter);
        while(i < k) {
            let maxKey = [...counter.keys()].reduce((a, b) =>
                counter.get(a) > counter.get(b) ? a : b
            );
            response.push(maxKey);
            counter.delete(maxKey);
            i++;
        } 
        return response;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numsSet = set(nums)
        maxFreq = 0
        for n in numsSet:
            if n - 1 not in numsSet:
                counter = 1
                while n + counter in numsSet:
                    counter += 1
                maxFreq = max(maxFreq, counter)

        return maxFreq
        